### PR TITLE
test: remove tmp file generated by test_thread_stat_io

### PR DESCRIPTION
**Pull request by rleungx**
## What have you changed? (mandatory)
Currently, the test named `test_thread_stat_io` in `tikv_util` will generate a temp file `theadnametest66` in the process of testing and the test itself doesn't remove it after finishing it. So another user runs this test will fail due to the permission.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:

- Bug fix 

## How has this PR been tested? (mandatory)

make dev

## Does this PR affect documentation (docs) or release note? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

File: components/tikv_util/src/metrics/threads_linux.rs [hunk 1]
```diff
@@ -314,6 +314,7 @@ mod tests {
                 f.sync_all().unwrap();
                 tx1.send(()).unwrap();
                 rx.recv().unwrap();
+                fs::remove_file(tmp).unwrap();
```
Comment (zhangjinpeng1987) [inline components/tikv_util/src/metrics/threads_linux.rs]: I'm curious why the directory generated by `temp_dir()` doesn't destroyed by it self when the test finished.

**Comment by siddontang** [comment]
PTAL @hicqu

File: components/tikv_util/src/metrics/threads_linux.rs (hunk 1 above)
Comment (siddontang) [reply components/tikv_util/src/metrics/threads_linux.rs]: same question here

**Comment by siddontang** [comment]
how do you find it? @rleungx

File: components/tikv_util/src/metrics/threads_linux.rs (hunk 1 above)
Comment (rleungx) [reply components/tikv_util/src/metrics/threads_linux.rs]: This `temp_dir` is from `std::env::temp_dir`.

**Comment by rleungx** [comment]
Just run `make dev` in our IDC and it fails because this file is owned by another guy. @siddontang

**Pull request merged**